ipam/calico-ipam.go

```diff
@@ -222,24 +222,41 @@ func cmdDel(args *skel.CmdArgs) error {
 		return err
 	}
 
-	// Release the IP address by using the handle - which is workloadID.
-	workloadID, _, err := utils.GetIdentifiers(args)
+	// Release the IP address by using the handleID.
+	handleID, err := utils.GetHandleID(conf.Name, args.ContainerID)
 	if err != nil {
 		return err
 	}
 
-	logger := utils.CreateContextLogger(workloadID)
+	logger := utils.CreateContextLogger(handleID)
 
-	logger.Info("Releasing address using workloadID")
-	if err := calicoClient.IPAM().ReleaseByHandle(workloadID); err != nil {
+	logger.Info("Releasing address using handleID")
+	if err := calicoClient.IPAM().ReleaseByHandle(handleID); err != nil {
 		if _, ok := err.(errors.ErrorResourceDoesNotExist); ok {
-			logger.WithField("workloadId", workloadID).Warn("Asked to release address but it doesn't exist. Ignoring")
-			return nil
+			logger.WithField("handleID", handleID).Warn("Asked to release address but it doesn't exist. Ignoring")
+		} else {
+			return err
 		}
+	} else {
+		logger.Info("Released address using handleID")
+	}
+
+	// Release the IP address by using the workloadID, to not leak IPs in existing installs.
+	workloadID, _, err := utils.GetIdentifiers(args)
+	if err != nil {
 		return err
 	}
+	workloadLogger := utils.CreateContextLogger(workloadID)
 
-	logger.Info("Released address using workloadID")
-	return nil
+	// We also need to handle the old way by workloadID, to not leak IPs in existing installs.
+	workloadLogger.Info("Releasing address using workloadID")
+	if err := calicoClient.IPAM().ReleaseByHandle(workloadID); err != nil {
+		if _, ok := err.(errors.ErrorResourceDoesNotExist); !ok {
+			return err
+		}
```

Comment:
And a debug log for the case where we get a `ResourceDoesNotExist` error.

Info is too high because this will be the mainline case, but debug should be fine and useful for debugging any potential issues with this code.